src/main/java/ca/corefacility/bioinformatics/irida/service/impl/sample/SampleServiceImpl.java

```diff
@@ -363,10 +378,28 @@ public Map<String, List<Long>> getSampleIdsBySampleNameForProjects(List<Long> pr
 	public void removeSequencingObjectFromSample(Sample sample, SequencingObject object) {
 		SampleSequencingObjectJoin readObjectForSample = ssoRepository.readObjectForSample(sample, object.getId());
 		ssoRepository.delete(readObjectForSample);
-		if (sample.getDefaultSequencingObject() != null
-				&& sample.getDefaultSequencingObject().getId().equals(object.getId())) {
+		if (sample.getDefaultSequencingObject() != null && sample.getDefaultSequencingObject()
+				.getId()
+				.equals(object.getId())) {
 			sampleRepository.removeDefaultSequencingObject(sample);
 		}
+		Set<AnalysisSubmission> submissions = submissionRepository.findAnalysisSubmissionsForSequencingObject(object);
+		if (submissions.isEmpty() && object.getSequencingRun() == null) {
+			Set<SequenceConcatenation> concatenations = concatenationRepository.findConcatenatedSequencingObjectSources(
+					object);
+			for (SequenceConcatenation concat : concatenations) {
+				concat.removeSource(object);
+				concatenationRepository.save(concat);
+			}
+			SequenceConcatenation concatenated = concatenationRepository.findConcatenatedSequencingObject(object);
```

Comment:
It might not be the concatenated object, but it could be a source for a concatenated object. I am of the understanding the sources have to be removed before the concatenated object.

Reply:
Ah ok! Well in that case can you just add a comment to above the code starting at line 388 to mention that?

Reply:
Please see e85217308c4cbfe1776876baa318d8b2fa2e4286.